.github/workflows/ci-unit-tests.yml

```diff
@@ -0,0 +1,58 @@
+name: Unit Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    env:
+      NODE_LTS: "10"
+    strategy:
+      matrix:
+        env:
+          - TEST_NODE_VERSION: "0.10"
+          - TEST_NODE_VERSION: "4"
+          - TEST_NODE_VERSION: "6"
+          - TEST_NODE_VERSION: "10"
+            LINT: "1"
+            COVER: "1"
+          - TEST_NODE_VERSION: "12"
+    name: unit-tests - node ${{ matrix.env.TEST_NODE_VERSION }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Configure Node Version (LTS)
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ env.NODE_LTS }}
+    - name: Install Base Project
+      run: |
+        npm install babel-cli
+        npm uninstall -D husky lint-staged
+        make node-modules
+        make build-node
+        rm -rf ${{ github.workspace }}/node_modules/ ${{ github.workspace }}/package-lock.json    
+    - name: Configure Node Version
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.env.TEST_NODE_VERSION }}
+      if: ${{ env.NODE_LTS != matrix.env.TEST_NODE_VERSION}}
+    - name: Install Dependencies
+      run: |
+        sudo sh -c 'echo 0 > /proc/sys/net/ipv6/conf/all/disable_ipv6'
+        node --version
+        npm install
+    - name: Run Linter
+      run: |
+        npm run lint
+      if: ${{ matrix.env.LINT == '1' }}
+    - name: Run Tests
+      run: |
+        make test-without-build
+    - name: Run Code Coverage
+      run: |
+        npm run cover && bash <(curl -s https://codecov.io/bash )
```

Comment:
better to use codecov action, see https://github.com/jaegertracing/jaeger/blob/master/.github/workflows/ci-unit-tests.yml#L25